File: exs/mundo_1/ruby/031.rb
=begin
Desafio 031

Problema: Desenvolva um programa que pergunte a distância de
          uma viagem em Km. Calcule o preço da passagem, cobrando
          R$0,50 por Km para viagens de até 200Km e R$0,45 para
          viagens mais longas.

Resolução do problema:
=end

print'Informe a distância em KM: '
d = gets.chomp.to_f

if d > 200
	preco = d * 0.45
else
	preco = d * 0.50
end

puts"O valor da passagem será R$ #{preco}."

File: exs/mundo_1/rust/031.rs
/*
Desafio 031

Problema: Desenvolva um programa que pergunte a distância de
          uma viagem em Km. Calcule o preço da passagem, cobrando
          R$0,50 por Km para viagens de até 200Km e R$0,45 para
          viagens mais longas.
*/

fn main() {
    let mut input = String::new();

    println!("Informe a distância em KM: ");
    std::io::stdin().read_line(&mut input).unwrap();

    let input = input.trim().parse::<f32>().expect("Input is not an number");

    let preco_passagem = if input > 200.0 {
        input * 0.45
    } else {
        input * 0.5
    };

    println!("O valor da passagem será R${:.2}", preco_passagem);
}
